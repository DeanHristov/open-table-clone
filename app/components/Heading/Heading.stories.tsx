import * as React from "react";
import { Meta, Story } from "@storybook/react";
import Heading, { IHeadingProps } from "./index";

export default {
  title: "Components/Heading",
  component: Heading,
} as Meta;

export const Default: Story<IHeadingProps> = (args) => <Heading {...args} />;

Default.args = {
  title: "Heading title",
  secondary: false,
};
