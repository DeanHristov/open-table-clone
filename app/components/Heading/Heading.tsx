"use client";

import { FC } from "react";
import classNames from "classnames";

export interface IHeadingProps {
  title: string;
  secondary?: boolean;
}

const Heading: FC<IHeadingProps> = ({ title, secondary }) => {
  const defaultStyles = [
    "text-neutral-8",
    "text-2xl",
    "font-bold",
    "leading-6",
  ];
  const styles = classNames(defaultStyles);

  return (
    <div className={`pb-4 ${secondary ? "border-b" : ""} border-gray-300`}>
      <h1 className={styles}>{title}</h1>
    </div>
  );
};

export default Heading;
