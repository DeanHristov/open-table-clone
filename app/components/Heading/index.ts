import Heading, { IHeadingProps } from "./Heading";

export type { IHeadingProps };
export default Heading;
