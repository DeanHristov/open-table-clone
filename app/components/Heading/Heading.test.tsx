import { render } from "@testing-library/react";
import Heading from "../Heading";

describe("Component: <Heading {...} />", () => {
  it("Should be able to render default state", () => {
    const { container } = render(<Heading title={"Some title"} />);

    expect(container).toMatchSnapshot();
  });

  it("Should be able to render the secondary state", () => {
    const { container } = render(<Heading secondary title={"Some title"} />);

    expect(container).toMatchSnapshot();
  });
});
