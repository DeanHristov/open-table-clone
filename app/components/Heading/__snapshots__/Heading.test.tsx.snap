// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Heading {...} /> Should be able to render default state 1`] = `
<div>
  <div
    class="pb-4  border-gray-300"
  >
    <h1
      class="text-neutral-8 text-2xl font-bold leading-6"
    >
      Some title
    </h1>
  </div>
</div>
`;

exports[`Component: <Heading {...} /> Should be able to render the secondary state 1`] = `
<div>
  <div
    class="pb-4 border-b border-gray-300"
  >
    <h1
      class="text-neutral-8 text-2xl font-bold leading-6"
    >
      Some title
    </h1>
  </div>
</div>
`;
